app/src/main/java/com/example/marvelapp/api/service/ServiceGenerator.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.marvelapp.api.service
+
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class ServiceGenerator {
+    private val httpClient = OkHttpClient.Builder().addInterceptor { chain ->
+        val defaultRequest = chain.request()
+        val defaultHttpUrl = defaultRequest.url()
+        val httpUrl = defaultHttpUrl.newBuilder()
+            .addQueryParameter(TS, TS_VALUE)
+            .addQueryParameter(KEY, KEY_VALUE)
+            .addQueryParameter(HASH, HASH_VALUE)
+            .build()
+
+        val requestBuilder = defaultRequest.newBuilder().url(httpUrl)
+
+        chain.proceed(requestBuilder.build())
+    }
+
+    private val builder = Retrofit.Builder()
+        .baseUrl(BASE_URL)
+        .addConverterFactory(GsonConverterFactory.create())
+
+    fun <S> createService(serviceClass: Class<S>): S {
+        val retrofit = builder.client(httpClient.build()).build()
+        return retrofit.create(serviceClass)
+    }
+
+    companion object {
```

Comment:
BASE_URL, KEY_VALUE, and HASH_VALUE should be declared in Gradle as a "buildConfigField"

Reply:
done